internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -191,6 +210,44 @@ func (r *BMACReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (c
 		return result.Result()
 	}
 
+	// Before reconciling agent spec, first check for the day2 flow
+	// if spoke cluster requirements are fulfilled.
+	fulfilled, clusterDeployment, err := r.spokeClusterRequirementsFulfilled(ctx, log, agent)
+	var spokeClient client.Client
+
+	if err != nil {
+		return reconcileError{err}.Result()
+	}
+
+	if fulfilled {
+		// Secret contains kubeconfig for the spoke cluster
+		secret := &corev1.Secret{}
```

Comment:
this logic should have it's own function as it should be used from ensureMCSCert and reconcileSpokeBMH

Reply:
I am not very sure about moving it into a common function. If still needed, we can discuss.